php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1538 @@
+          b.token(PHPLexicalGrammar.BINARY)),
+        b.token(PHPPunctuator.RPARENTHESIS),
+        UNARY_EXPR()));
+  }
+
+  public ExpressionTree UNARY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.UNARY_EXPR).is(
+      b.firstOf(
+        f.prefixExpr(
+          b.firstOf(
+            b.token(PHPPunctuator.INC),
+            b.token(PHPPunctuator.DEC),
+            b.token(PHPPunctuator.PLUS),
+            b.token(PHPPunctuator.MINUS),
+            b.token(PHPPunctuator.TILDA),
+            b.token(PHPPunctuator.BANG),
+            b.token(PHPPunctuator.AT)),
+          UNARY_EXPR()),
+        CAST_EXPR(),
+        ASSIGNMENT_EXPRESSION(),
+        ASSIGNMENT_BY_REFERENCE(),
+        POSTFIX_EXPRESSION()));
+  }
+
+  public ExpressionTree CONCATENATION_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.CONCATENATION).is(
+      f.concatenationExpr(
+        UNARY_EXPR(),
+        b.zeroOrMore(f.newTuple60(
+          b.token(DOT),
+          UNARY_EXPR()))));
+  }
+
+  public ExpressionTree MULTIPLICATIVE_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.MULTIPLICATIVE_EXPR).is(
+      f.multiplicativeExpr(
+        CONCATENATION_EXPR(),
+        b.zeroOrMore(f.newTuple61(
+          b.firstOf(b.token(STAR), b.token(DIV), b.token(MOD)),
+          CONCATENATION_EXPR()))));
+  }
+
+  public ExpressionTree ADDITIVE_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ADDITIVE_EXPR).is(
+      f.additiveExpr(
+        MULTIPLICATIVE_EXPR(),
+        b.zeroOrMore(f.newTuple62(
+          b.firstOf(b.token(PLUS), b.token(MINUS)),
+          MULTIPLICATIVE_EXPR()))));
+  }
+
+  public ExpressionTree SHIFT_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SHIFT_EXPR).is(
+      f.shiftExpr(
+        ADDITIVE_EXPR(),
+        b.zeroOrMore(f.newTuple63(
+          b.firstOf(b.token(SL), b.token(SR)),
+          ADDITIVE_EXPR()))));
+  }
+
+  public ExpressionTree RELATIONAL_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.RELATIONAL_EXPR).is(
+      f.relationalExpr(
+        SHIFT_EXPR(),
+        b.zeroOrMore(f.newTuple64(
+          b.firstOf(b.token(LE), b.token(GE), b.token(LT), b.token(GT)),
+          SHIFT_EXPR()))));
+  }
+
+  public ExpressionTree EQUALITY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EQUALITY_EXPR).is(
+      f.equalityExpr(
+        RELATIONAL_EXPR(),
+        b.zeroOrMore(f.newTuple65(
+          b.firstOf(b.token(NOTEQUAL2), b.token(NOTEQUAL), b.token(EQUAL2), b.token(EQUAL), b.token(NOTEQUALBIS)),
+          RELATIONAL_EXPR()))));
+  }
+
+  public ExpressionTree BITWISE_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.BITWISE_AND).is(
+      f.bitwiseAndExpr(
+        EQUALITY_EXPR(),
+        b.zeroOrMore(f.newTuple66(
+          b.token(PHPPunctuator.AMPERSAND),
+          EQUALITY_EXPR()))));
+  }
+
+  public ExpressionTree BITWISE_XOR_EXPR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)